test/data/test-suite-data.json

```diff
@@ -122,7 +122,7 @@
   {
     "description": "docker uses qualifiers and hash image id as versions",
     "purl": "pkg:docker/customer/dockerimage@sha256:244fd47e07d1004f0aed9c?repository_url=gcr.io",
-    "canonical_purl": "pkg:docker/customer/dockerimage@sha256:244fd47e07d1004f0aed9c?repository_url=gcr.io",
+    "canonical_purl": "pkg:docker/customer/dockerimage@sha256%3A244fd47e07d1004f0aed9c?repository_url=gcr.io",
```

Comment:
The official test input says this should be unencoded (':').

I recommend keeping test-suite-data.json exactly the same as in the spec repository and making a separate file for tests that are unique to this repository. That makes it easier to ensure this implementation matches the spec, easier to update the copy of the test suite file, and easier to find tests that could be added to the spec.

Reply:
Good point.